File: gommo-frontend/src/modules/collaborator/lib/collaborator.mapper.ts
import type {Collaborator, CollaboratorCreateDto} from "@/modules/collaborator/dto/collaborator.dto";
import {digitsOnly} from "@/shared/lib/input/digits";
import {unmaskRg} from "@/shared/lib/input/rg";

export function collaboratorToFormDto(collaborator: Collaborator): CollaboratorCreateDto {
    return {
        fullName: collaborator.fullName,
        socialName: collaborator.socialName,
        cpf: digitsOnly(collaborator.cpf),
        rg: collaborator.rg ? unmaskRg(collaborator.rg) : collaborator.rg,
        birthDate: collaborator.birthDate?.slice(0, 10) ?? "",
        gender: collaborator.gender,
        maritalStatus: collaborator.maritalStatus,
        motherName: collaborator.motherName,
        fatherName: collaborator.fatherName,
        nationality: collaborator.nationality,
        pisPasep: collaborator.pisPasep,
    };
}

export const emptyCollaboratorForm = (): CollaboratorCreateDto => ({
    fullName: "",
    cpf: "",
    birthDate: "",
});

File: gommo-backend/src/main/java/br/com/gommo/modules/collaborator/mapper/CollaboratorMapper.java
package br.com.gommo.modules.collaborator.mapper;

import br.com.gommo.modules.collaborator.dto.CollaboratorRequestDto;
import br.com.gommo.modules.collaborator.dto.CollaboratorResponseDto;
import br.com.gommo.modules.collaborator.entity.Collaborator;
import org.springframework.stereotype.Component;

@Component
public class CollaboratorMapper {

    public Collaborator toEntity(CollaboratorRequestDto dto) {
        return Collaborator.builder()
                .fullName(dto.getFullName())
                .socialName(dto.getSocialName())
                .cpf(dto.getCpf())
                .rg(dto.getRg())
                .birthDate(dto.getBirthDate())
                .gender(dto.getGender())
                .maritalStatus(dto.getMaritalStatus())
                .motherName(dto.getMotherName())
                .fatherName(dto.getFatherName())
                .nationality(dto.getNationality() != null ? dto.getNationality() : "Brasileiro")
                .pisPasep(dto.getPisPasep())
                .build();
    }

    public void updateEntity(Collaborator entity, CollaboratorRequestDto dto) {
        entity.setFullName(dto.getFullName());
        entity.setSocialName(dto.getSocialName());
        entity.setCpf(dto.getCpf());
        entity.setRg(dto.getRg());
        entity.setBirthDate(dto.getBirthDate());
        entity.setGender(dto.getGender());
        entity.setMaritalStatus(dto.getMaritalStatus());
        entity.setMotherName(dto.getMotherName());
        entity.setFatherName(dto.getFatherName());
        if (dto.getNationality() != null) {
            entity.setNationality(dto.getNationality());
        }
        entity.setPisPasep(dto.getPisPasep());
    }

    public CollaboratorResponseDto toResponse(Collaborator entity) {
        return CollaboratorResponseDto.builder()
                .id(entity.getId())
                .status(entity.getStatus())
                .fullName(entity.getFullName())
                .socialName(entity.getSocialName())
                .cpf(entity.getCpf())
                .rg(entity.getRg())
                .birthDate(entity.getBirthDate())
                .gender(entity.getGender())
                .maritalStatus(entity.getMaritalStatus())
                .motherName(entity.getMotherName())
                .fatherName(entity.getFatherName())
                .nationality(entity.getNationality())
                .pisPasep(entity.getPisPasep())
                .createdAt(entity.getCreatedAt())
                .updatedAt(entity.getUpdatedAt())
                .build();
    }
}
